### PR TITLE
Rebuild for python310 (v2)

### PR DESCRIPTION
**Pull request by duncanmmacleod**
This PR is a duplicate of #30 with the extra patch from https://github.com/conda-forge/pyepics-feedstock/pull/30#issuecomment-984504224.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/package_data.patch [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ """
+ 
+ 
++# determine which libca / libCom dll is appropriate
++import platform
++mach = platform.machine()
++if mach.startswith('arm'):
++    nbits = 'arm'
++else:
++    nbits = platform.architecture()[0].replace('bit', '')
```
Comment (mrakitin) [inline recipe/package_data.patch:15]: For future us, this is the output of the command ([docs](https://docs.python.org/3/library/platform.html#platform.architecture)):
```py
In [1]: import platform

In [2]: platform.architecture()
Out[2]: ('64bit', 'ELF')
```
so, the first element is taken, and the `bit` string is stripped from that.
:+1:

File: recipe/package_data.patch [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
++if os.name == 'nt':
++    libsrc = 'win'
++elif sys.platform == 'darwin':
++    libsrc = 'darwin'
++elif sys.platform.startswith('linux'):
++    libsrc = 'linux'
```
Comment (mrakitin) [inline recipe/package_data.patch:21]: It uses two different ways to get the platform. Can it be handled via the earlier used `platform`, such as [`platform.system()`](https://docs.python.org/3/library/platform.html#platform.system), that returns one of 'Linux', 'Darwin', 'Java', 'Windows' (needs to be lower-cased and the `win` case has to be handled specially)? Just not very happy with the `os`, `sys`, and `platform` used to determine it.

**Review by mrakitin — COMMENTED** [review]
Looks great to me, thanks, @duncanmmacleod!

I left a comment about the way to determine the platform, but the current way should work OK. If you have a chance to update it, I'd appreciate it. Thanks again.

File: recipe/package_data.patch (hunk 2 above)
Comment (duncanmmacleod) [reply recipe/package_data.patch:21]: I copied this code essentially verbatim from the library itself, but have updated the patch in line with your comments, please rereview.

Comment (mrakitin) [reply recipe/package_data.patch:21]: I do not see the changes here. Was the commit puahed?

Comment (duncanmmacleod) [reply recipe/package_data.patch]: No. No it wasn't, sigh; sorry. It is there now.

**Review by mrakitin — APPROVED** [review]
I am happy with the change.

@tacaswell, do you have any concerns/comments?

**Comment by mrakitin** [comment]
Looks like we have the same problem with https://github.com/conda-forge/xraylarch-feedstock/pull/4. The [`setup.py`](https://github.com/xraypy/xraylarch/blob/0c376a31f057a066ae15976d5f7215e96ac47b91/setup.py#L119-L121) has the same issues with all-platforms `package_data`.

Thanks for the pointers, @duncanmmacleod, I'll apply a similar patch there :slightly_smiling_face:

**Comment by mrakitin** [comment]
By the way, the trick worked for https://github.com/conda-forge/xraylarch-feedstock/pull/4!

**Comment by duncanmmacleod** [comment]
@conda-forge/pyepics can this be merged now?

**Comment by tacaswell** [comment]
> tacaswell, do you have any concerns/comments?

A bit late, but yes.  We should not be including _any_ of the pre-built binaries and instead be using the epics binaries provided by the epics-base package.

**Pull request merged**